src/System.Linq.Expressions/src/System/Dynamic/DynamicMetaObject.cs

```diff
@@ -261,7 +262,7 @@ internal static Expression[] GetExpressions(DynamicMetaObject[] objects)
                 DynamicMetaObject mo = objects[i];
                 ContractUtils.RequiresNotNull(mo, nameof(objects));
                 Expression expr = mo.Expression;
-                ContractUtils.RequiresNotNull(expr, nameof(objects));
+                Debug.Assert(expr != null, "Unexpected null expression; ctor should have caught this.");
```

Comment:
`Expression` is a read-only property which gets set in the constructor which performs a `Contracts.RequiresNotNull` check on the parameter that gets passed.